### DISCUSSION TITLE
Other implementations

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-cli-restsource#186] [Ideas]
I met with a customer today who is using another product and had interest in integrations.

I wanted to start a discussion to help cross link those requests to see if other customers are interested in products supporting WinGet.

**Comment by denelon** [comment]
JFrog
https://www.jfrog.com/jira/browse/RTFACT-22211

**Comment by byjrack** [comment]
I know this is ancient, but stumbled across it and figured why not

- Love to have a source that worked with a completely static site like formulae.brew.sh 
- Git repository as a source In the spirit of a `brew tap`.
- Intune tenant Apps as Source.  If we put something in Company Portal allow winget to pull  it or a bundle of things via `import`
- Tools like Flexera AdminStudio have a record of the app and metadata and may be able to provide enough for winget 

The installer and the metadata for winget don't need to be collocated much like the existing community repository.  Having both metadata and package in the same endpoint are helpful (intune and maybe arti) but lots of options for hosting the packages that work with a GET.

**Reply by denelon** [reply]
Thanks for the suggestions! We're getting ready to focus a bit more on WinGet sources, so this is timely feedback.

**Reply by byjrack** [reply]
Glad to hear.  I had been poking around the index.db and toyed with preindexing.  Not as granular with the pathparts and seeing what else I could slack off with while Winget being functional.  Given time and lots of trial and error I could probably make it work where I could build a db from a repo of yamls, but much rather pull something off the shelf.

**Reply by byjrack** [reply]
Any public roadmap items on possible alternative implementations we can watch out for?  Keep stumbling across situations where having a curated or tweaked set of yamls would be super helpful.  Going full restsource feels like overkill so getting to static or git backed with authentication baked in will be super helpful.

**Comment by denelon** [comment]
[Configure Nerdio Manager for Unified Application Management](https://nmw.zendesk.com/hc/en-us/articles/14122129771031-Configure-Nerdio-Manager-for-Unified-Application-Management)

**Reply by byjrack** [reply]
Yeah we are an Intune shop so that won't change anytime soon.

**Comment by denelon** [comment]
[winget.Pro](https://winget.pro/)

**Reply by byjrack** [reply]
yeah AGPL is not ideal for us and creating a new supplier relationship would be more complicated than me just creating an MSIX pipeline for myself.

**Comment by byjrack** [comment]
So should I read into the suggestions as nothing really planned for first-party alternatives to the official resource option and the DIY MSIX of sqllite and YAMLs?

**Reply by denelon** [reply]
We're still evaluating the relative priority for different implementations of the REST source. Most of the requests we're currently receiving are suited well for hosting a source on Azure. Be sure to add your 👍 to:
* https://github.com/microsoft/winget-cli-restsource/issues/158

**Reply by byjrack** [reply]
Yup for me its the overhead of the app.  we can do app services in some cases, but its pretty low usage in my org.  I don't mind azure biased for the resources, but hoping it can be more ephemeral service backed by cosmos.    a helm chart would also tick the box, but that may get into the neutral answer camp. gitops for static content for me is still the easiest path to self-managed.

I did try and talk to our intune contacts about an endpoint that is backed by intune's app registrations and syndication from community (similar to Store) but no interest.  They already built a lot of the winget/packagemanager interop into agentexecutor and while it may not tick every box (user/system context) it can hit a lot of the common ones.
